cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/apps/map.js

```diff
@@ -0,0 +1,185 @@
+  store.subscribe( this.renderCounties.bind( this ) );
+};
+
+MortgagePerformanceMap.prototype.onClick = function( event ) {
+  var change = new Event( 'change' );
+  this.$container.querySelector( 'input[name="mp-map_geo"]:checked' ).checked = false;
+  this.$form.dispatchEvent( change );
+  event.preventDefault();
+};
+
+MortgagePerformanceMap.prototype.onChange = function( event ) {
+  var action, geoEl, geoType, geoId, geoName, countyState, date;
+
+  switch ( event.target.id ) {
+    case 'mp-map_geo-state':
+    case 'mp-map_geo-metro':
+    case 'mp-map_geo-county':
+      geoType = this.$container.querySelector( 'input[name="mp-map_geo"]:checked' ).id.replace( 'mp-map_geo-', '' );
+      geoId = '';
+      geoName = '';
+      action = actions.setGeo( geoId, geoName, geoType );
+      break;
+    case 'mp-map-state':
+      geoType = this.$container.querySelector( 'input[name="mp-map_geo"]:checked' ).id.replace( 'mp-map_geo-', '' );
+      if ( geoType === 'state' ) {
+        geoId = this.$state.value;
+        geoName = this.$state.options[this.$state.selectedIndex].text;
+        action = actions.updateChart( geoId, geoName, geoType );
+      } else {
+        var abbr = this.$state.options[this.$state.selectedIndex].getAttribute( 'data-abbr' );
+        if ( !abbr ) {
+          return this.chart.highchart.chart.zoomOut();
+        }
+        action = actions.fetchCounties( abbr );
+      }
+      break;
+    case 'mp-map-metro':
+      geoId = this.$metro.value;
+      geoName = this.$metro.options[this.$metro.selectedIndex].text;
+      action = actions.updateChart( geoId, geoName );
+      break;
+    case 'mp-map-county':
+      geoId = this.$county.value;
+      geoName = this.$county.options[this.$county.selectedIndex].text;
+      action = actions.updateChart( geoId, geoName );
+      break;
+    case 'mp-map-month':
+    case 'mp-map-year':
+      date = `${ this.$year.value }-${ this.$month.value }`;
+      action = actions.updateDate( date );
+      break;
+    default:
+      action = actions.clearGeo();
+  }
+
+  store.dispatch( action );
+
+};
+
+MortgagePerformanceMap.prototype.zoom = function( prevState, state ) {
+  this.chart.highchart.chart.get( state.geo.id ).zoomTo();
+};
+
+MortgagePerformanceMap.prototype.renderChart = function( prevState, state ) {
+  if ( !state.geo.id ) {
+    this.chart.highchart.chart.zoomOut();
+  }
+  if ( prevState.date === state.date && prevState.geo.type === state.geo.type && state.geo.id ) {
+    this.chart.highchart.chart.get( state.geo.id ).select();
+    this.chart.highchart.chart.get( state.geo.id ).zoomTo();
+    this.chart.highchart.chart.mapZoom( 5 );
+  }
+  if ( prevState.date !== state.date || prevState.geo.type !== state.geo.type ) {
+    store.dispatch( actions.startLoading() );
+    this.chart.update( {
+      source: `map-data/${ this.timespan }/${ _plurals[state.geo.type] }/${ state.date }`,
+      metadata: _plurals[state.geo.type]
+    } ).then( () => {
```

Comment:
how does Babel handle promises?

Reply:
Mmmm good question. I know cfpb-chart-builder uses promises and it supports IE9+. I'll do some investigating before merging this PR.